pkg/reconciler/broker/broker.go

```diff
@@ -205,13 +209,33 @@ func (r *Reconciler) reconcileBroker(ctx context.Context, b *brokerv1beta1.Broke
 		} else {
 			m.SetState(config.State_UNKNOWN)
 		}
-	})
 
-	// Update config map
-	// TODO should this happen in a defer so there's an update regardless of
-	// error status, or only when reconcile is successful?
-	r.flagTargetsForUpdate()
-	return nil
+		// Insert each Trigger to the config.
+		for _, t := range triggers {
+			if t.Spec.Broker == b.Name {
+				target := &config.Target{
+					Id:        string(t.UID),
+					Name:      t.Name,
+					Namespace: t.Namespace,
+					Broker:    b.Name,
+					Address:   t.Status.SubscriberURI.String(),
+					RetryQueue: &config.Queue{
+						Topic:        resources.GenerateRetryTopicName(t),
+						Subscription: resources.GenerateRetrySubscriptionName(t),
+					},
```

Comment:
At this stage, are we sure the trigger retry topic/sub exist? Unlike when we have broker/trigger reconcilers together, this might no longer hold true, right? Is there a way to tell from the trigger status (I remember Grant added custom trigger status for gcp broker triggers).

Reply:
I care because the retry service will keep trying to connect to the retry sub.

Reply:
Yeah ideally we should get the pubsub resources from broker/trigger status. Unfortunately the custom status doesn't work today due to webhook issues. This is yet another reason to prioritize custom trigger status.

I believe the retry will eventually work once the pubsub resources exist, but let me verify that.  Given that this should happen rarely in practice, we can work on it as a follow up.

@yolocs also pointed out that retry can check `target.State` field and should not establish pubsub connection if trigger is not ready.

Reply:
I tested retry and it will eventually recover once the pubsub topic/sub exist. #1154 to track the improvements that @yolocs proposed.

Reply:
The custom status fields would only indicate the resource names, not the readiness. They are not necessary here because the reconcilers updating the configmap and creating the resources will always be in the same container, so they can use the same name generation function. Not perfect encapsulation, but good enough for now.